akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.projection.cassandra.internal
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.Promise
+import scala.concurrent.duration.FiniteDuration
+
+import akka.Done
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.InternalApi
+import akka.projection.Projection
+import akka.projection.ProjectionId
+import akka.stream.KillSwitches
+import akka.stream.alpakka.cassandra.scaladsl.CassandraSessionRegistry
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class CassandraProjectionImpl[Offset, StreamElement](
+    override val projectionId: ProjectionId,
+    sourceProvider: Option[Offset] => Source[StreamElement, _],
+    offsetExtractor: StreamElement => Offset,
+    saveOffsetAfterNumberOfEvents: Int,
```

Comment:
TODO: rename to saveOffsetAfterNumberOfElements (or saveOffsetAfterElements?)

Reply:
`saveOffsetElementInterval`?